### PR TITLE
don't access course id unless the product is a seat

### PR DESCRIPTION
**Pull request by McKenzieW**
Saw NoneType errors in stage when I tried to apply new coupon I made to an entitlement product

[LEARNER-4166](https://openedx.atlassian.net/browse/LEARNER-4166), [Ecommerce](https://openedx.atlassian.net/browse/LEARNER-4359)

### CODE REVIEW

**Review by staubina — APPROVED** [review]

File: ecommerce/extensions/offer/models.py [hunk 1]
```diff
@@ -64,22 +64,25 @@ def _identify_uncached_product_identifiers(self, lines, domain, partner_code, qu
         course_uuids = []
         applicable_lines = lines
         for line in applicable_lines:
-            product = line.product
+            if line.product.is_seat_product:
+                product_id = line.product.course.id
+            else:
```
Comment (Mjloturco) [inline ecommerce/extensions/offer/models.py]: maybe worth giving this an explicit elif? again trying to think forward for if we have additional product types at somepoint.

**Review by Mjloturco — CHANGES_REQUESTED** [review]
Might want to add a bit more defensiveness if we are in here anyway.

File: ecommerce/extensions/offer/models.py (hunk 1 above)
Comment (McKenzieW) [reply ecommerce/extensions/offer/models.py]: good call

**Comment by Mjloturco** [comment]
Nice, do we want to do something in the case that it is not a Seat or an Entitlement? we could throw an error or log or maybe doing nothing is the correct call for that case, I'm unsure.

**Comment by McKenzieW** [comment]
@Mjloturco yes we definitely do, the product should be removed from applicable lines. Such a keen eye you have.

**Comment by McKenzieW** [comment]
So the `_filter_for_paid_course_products` method already removes any products that aren't seats or or course entitlements from the applicable lines. @Mjloturco do you think it's still important to handle products falling outside of that category in `_identify_uncached_product_identifiers`, or will it be sufficient to just more verbosely describe the `else` block with an `elif...`?

**Review by Mjloturco — APPROVED** [review]

**Comment by Mjloturco** [comment]
@McKenzieW nice, I didn't have the context that other products would already be filtered out, Looks good to me (once you can convince codecov to stop complaining)

**Pull request merged**